### PR TITLE
support R 4.0 raw string search

### PR DESCRIPTION
**Pull request by randy3k**
TODO:

- [x] write tests
- [ ] ~~better multiline support~~

### CODE REVIEW

File: src/fsm.c [hunk 1]
```diff
@@ -6,24 +6,106 @@ typedef struct {
...
+                state->double_quoted = 1;
+                state->raw_dashes_temp = 0;
+                return;
+            } else {
+                state->raw_parse_state = 0;
+                state->double_quoted = 0;
+                state->single_quoted = 0;
+            }
+        } else if (state->raw_parse_state == 2) {
+            if (c == '(' || c == '[' || c == '{') {
```
Comment (renkun-ken) [inline src/fsm.c:42]: Looks like `|` works too. 

See https://github.com/wch/r-source/blob/1977750ab278e6e3ea5c29b6a1916c7398e9e2fa/src/main/gram.c#L5038.

**Comment by randy3k** [comment]
I didn't have time to work on this, so I guess we will skip it for v0.3.7

File: src/fsm.c (hunk 1 above)
Comment (randy3k) [reply src/fsm.c:42]: Do you think we should also support `|`?

**Comment by randy3k** [comment]
I have decided not to improve multiline support in this PR but rather to merge it first.

After merging this PR, I reckon we should freeze the master branch and prepare a release.

File: src/fsm.c (hunk 1 above)
Comment (renkun-ken) [reply src/fsm.c:42]: I guess we should not since it is undocumented behavior.

**Pull request merged**